web-common/src/features/entity-management/resource-invalidations.ts

```diff
@@ -82,6 +82,16 @@ async function invalidateResource(
   refreshResource(queryClient, instanceId, resource);
 
   const lastStateUpdatedOn = getLastStateUpdatedOn(resource);
+  if (
+    resource.meta.reconcileStatus !== V1ReconcileStatus.RECONCILE_STATUS_IDLE &&
+    !lastStateUpdatedOn
+  ) {
+    // When a resource is created it can send an event with status = IDLE just before it is queued for reconcile.
+    // So handle the case when it is 1st queued and status != IDLE
+    resourcesStore.setVersion(resource);
```

Comment:
Nope this is because the initial resource created has status=idle. So it is not easy to differentiate between a fresh resource vs a resource that became status=idle. This piece of code ignores such cases.

Reply:
I am hoping this is a temporary fix until we streamline reconcile.

Reply:
Ok I understand that when a resource is created, its first status is `IDLE`. But I still don't understand why `resourcesStore.setVersion(resource)` fixes the bug.

Maybe one thing that could help the readability is well-named booleans that clarify the significance of `!lastStateUpdatedOn` and `lastStateUpdatedOn === resource.meta.stateUpdatedOn`. Like:
```typescript
const isNewResource = !lastStateUpdatedOn
const whatDoesThisMean = lastStateUpdatedOn === resource.meta.stateUpdatedOn`
```

Reply:
Ya I hear you. Lets fix this is in a better way once the release is complete.